.github/workflows/release.yaml

```
@@ -0,0 +1,62 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to be released'
+        required: true
+
+permissions:
+  packages: write
+  id-token: write
+  contents: read
+
+jobs:
+  publish:
+    name: preliminary-checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Ensure version is properly set
+        run: |
+          test "$(yarn -s run get-version)" = "${{ inputs.tag }}"
+      - name: Give user instruction
+        if: failure()
+        run: echo "::error file=package.json,line=$(sed -n '/"version":/=' package.json)::The tag ${{ inputs.tag }} must match the version $(yarn -s run get-version)"
```

probably cleaner (and more readable) as a single step, since the two are related
(and use environment to pass input, to ensure they are sanitized)

```suggestion
      - name: Ensure version is properly set
        run: |
           PKG_VERSION="$(yarn -s run get-version)"
           if ! [ "$PKG_VERSION" = "$VERSION" ]; then
             echo "::error file=package.json,line=$(sed -n '/"version":/=' package.json)::The tag $VERSION must match the version $PKG_VERSION specified in package.json"
             exit 1
          fi
        env:
           VERSION: ${{ inputs.tag }}
```